Firmware/Marlin_main.cpp

```diff
@@ -3435,28 +3435,31 @@ static void gcode_M600(const bool automatic, const float x_position, const float
     plan_buffer_line_curposXYZE(FILAMENTCHANGE_XYFEED);
     st_synchronize();
 
-    // Unload filament
-    if (MMU2::mmu2.Enabled()) {
-        eject_slot = MMU2::mmu2.get_current_tool();
-        mmu_M600_unload_filament();
-    } else {
-        // Beep, manage nozzle heater and wait for user to start unload filament
-        M600_wait_for_user();
-        unload_filament(e_shift_late);
-    }
-    st_synchronize();          // finish moves
-    {
+    boolean repeat = false;
+    do {
+        // Unload filament
+        if (MMU2::mmu2.Enabled()) {
+            eject_slot = MMU2::mmu2.get_current_tool();
+            mmu_M600_unload_filament();
+        } else {
+            // Beep, manage nozzle heater and wait for user to start unload filament
+            M600_wait_for_user();
+            unload_filament(e_shift_late);
+        }
+        st_synchronize();          // finish moves
+
         FSensorBlockRunout fsBlockRunout;
         
         if (!MMU2::mmu2.Enabled())
         {
             KEEPALIVE_STATE(PAUSED_FOR_USER);
             uint8_t choice =
                 lcd_show_fullscreen_message_yes_no_and_wait_P(_i("Was filament unload successful?"), false, LCD_LEFT_BUTTON_CHOICE); ////MSG_UNLOAD_SUCCESSFUL c=20 r=3
+            lcd_update_enable(false);
             if (choice == LCD_MIDDLE_BUTTON_CHOICE) {
                 lcd_clear();
                 lcd_puts_at_P(0, 2, _T(MSG_PLEASE_WAIT));
-                current_position[X_AXIS] -= 100;
+                current_position[X_AXIS] = 100;
```

Comment:
Why did you change that?

Reply:
If I repeatedly choose the No answer at the question "Was filament unload successful?", the print head moves to the left again and again, till it crashes to the left edge.